components/query.py

```diff
@@ -150,5 +181,7 @@ def get_filenames(df, subspecies, view, sex, hybrid, num_images):
         filepaths = df_filtered.file_url.astype('string').values
         #return list of filenames for min(user-selected, available) images randomly selected images from the filtered dataset
         return list(filenames), list(filepaths)
+    elif missing_vals > 0:
+        return [0, missing_vals], 0
     else:
-        return 0, 0
+        return [0], 0
```

Comment:
I find this function's return value confusing. When everything works two arrays of strings are returned. When there are no filenames we return `[0],0` - an array of numbers and a single number. The `[0, missing_vals], 0` return value seems like an unexpected way to return an error/warning. Perhaps you could just raise an exception and catch it in the calling function. To me it seems like this function should always return two arrays of strings.

Reply:
So I did have an unnecessary `else`, in that just having 
```
if max_imgs > 0:
    ....
else:
return [0, missing_vals], 0
```
would be sufficient for what I'm trying to accomplish it alerting the user to the fact that entries exist, just not images or that there aren't any. 

However, maybe that's trying to do too much with this and it should just be filtered within `get_images` to return such strings or call `get_filename`?

Reply:
This is still confusing to me. Having a magic value of 0 in the first slot in the first array communicating that the second value in the array is the number of missing values is confusing. This seems like an error being hidden in a return value.  Also the second array changes into a number of 0 is unexpected. Raising an exception and having a consistent return type is a more typical approach to this kind of problem.

If you really want to return the number of missing values instead of raising an exception I would add it as a third value returned.